src/components/NewAnalysisModal.js

```diff
@@ -64,33 +139,120 @@ export const NewAnalysisModal = Utils.withDisplayName('NewAnalysisModal')(
+      div({ style: { opacity: currentApp ? '.5' : '1', ...toolCardStyles }, onClick: () => { setCurrentTool(tools.galaxy.label); setNextViewMode(tools.galaxy.label) }, disabled: !currentApp, title: currentApp ? 'You already have a galaxy environment' : '' }, [img({ src: galaxyLogo, style: _.merge(imageStyles, { width: '30%' }) })])
     ])
 
-    const renderCreateAnalysisBody = () => div({ style: { display: 'flex', flexDirection: 'column', flex: 1, padding: '.5rem 1.5rem 1.5rem 1.5rem' } }, [
-      renderToolButtons()
+    const renderSelectAnalysisBody = () => div({ style: { display: 'flex', flexDirection: 'column', flex: 1, padding: '.5rem 1.5rem 1.5rem 1.5rem' } }, [
+      renderToolButtons(),
+      h(Dropzone, {
+        accept: `.${tools.Jupyter.ext}, .${tools.RStudio.ext}`,
+        style: { flexGrow: 1, backgroundColor: colors.light(), height: '100%' },
+        activeStyle: { backgroundColor: colors.accent(0.2), cursor: 'copy' },
+        onDropRejected: () => reportError('Not a valid analysis file',
+          'The selected file is not a .ipynb otebook file or an .Rmd rstudio file. Ensure your file has the proper extension.'),
+        onDropAccepted: files => {
+          const tool = getTool(files.pop().path)
+          setCurrentTool(tool)
+          currentRuntime && !isRuntimeDeletable(currentRuntime) && currentRuntimeTool !== tool ? onSuccess() : setNextViewMode(tool, NEW_ANALYSIS_MODE)
+          uploadFiles()
+        }
+      }, [() => div({
+        onClick: () => {
+          resetView()
+          onSuccess()
+          openUploader()
+        }, style: { marginTop: '1rem', fontSize: 16, lineHeight: '20px', ...Style.elements.card.container, alignItems: 'center', width: '100%', height: 150, backgroundColor: colors.dark(0.1), border: `1px dashed ${colors.dark(0.7)}`, boxShadow: 'none' }
+      }, [
+        div(['Or Click / Drag to upload an analysis file']),
+        icon('upload-cloud', { size: 25, style: { opacity: 0.4, marginTop: '0.5rem' } })
+      ])])
+    ])
+
+    const getArtifactLabel = toolLabel => Utils.switchCase(toolLabel, [tools.RStudio.label, () => 'R markdown file'],
+      [tools.Jupyter.label, () => 'notebook'],
+      [Utils.DEFAULT, () => console.error(`Should not be calling getArtifactLabel for ${toolLabel}, artifacts not implemented`)])
+
+    const renderCreateAnalysis = () => div({ style: { display: 'flex', flexDirection: 'column', flex: 1, padding: '.5rem 1.5rem 1.5rem 1.5rem' } }, [
+      h2({ style: { fontWeight: 600, marginBottom: 0 } }, [`Create a new ${getArtifactLabel(currentTool)}`]),
+      renderCreateAnalysisBody(currentTool)
+    ])
+
+    const existingNames = _.map(({ name }) => getDisplayName(name), analyses)
+
+    const errors = validate(
+      { analysisName, notebookKernel },
+      {
+        analysisName: analysisNameValidator(existingNames),
```

Comment:
Since this is used once, in-lining may be cleaner 
```suggestion
        analysisName: _.map(({ name }) => getDisplayName(name), analyses),
```